compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -831,8 +831,16 @@ abstract class GenJSCode extends plugins.PluginComponent
           }
 
           val call = genApplyMethod(genThis(), sym, jsParams.map(_.ref))
-          val body = ensureBoxed(call,
-              enteringPhase(currentRun.posterasurePhase)(sym.tpe.resultType))
+          val resTpeEnteringPosterasure = enteringPhase(currentRun.posterasurePhase) {
+            sym.tpe match {
+              case _: ExistentialType =>
+                // Workaround for #1581
```

Comment:
Please describe the issue in here shortly.
(something like: "We should not see an ExistentialType here. This is a Scalac 2.10 bug. We assume no boxing is required. See #1581").